vespalib/src/vespa/vespalib/stllike/hash_map.h

```diff
@@ -50,6 +50,25 @@ class hash_map
     void erase(const_iterator it)               { return erase(it->first); }
     iterator find(const K & key)                { return _ht.find(key); }
     const_iterator find(const K & key)    const { return _ht.find(key); }
+
+    template< typename AltKey, typename AltExtract=std::_Identity<K>, typename AltHash=vespalib::hash<AltKey>, typename AltEqual=equal_to<AltKey, K> >
+    const_iterator find(const AltKey & key) const {
```

Comment:
Boost.MultiIndex also supports heterogenous lookups in a slightly different way by explicitly specifying hash and equality functors at `find()`-time: https://www.boost.org/doc/libs/1_68_0/libs/multi_index/doc/tutorial/indices.html#hash_lookup

Again the supported types must be specified directly to avoid type conversion issues.

Personally I'm leaning rather more towards the Working Group proposal above since it automatically works for all invocations of `find()` and only specifies the hashing/equality policy once.

Reply:
I agree. Past experiences with boost does not run in their favour.